src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -225,7 +225,17 @@ class ArnoldOutput : public IECore::RefCounted
 				AiNodeSetStr( m_driver.get(), AiParamGetName( fileNameParameter ), output->getName().c_str() );
 			}
 
-			IECore::StringVectorDataPtr customAttributesData = new IECore::StringVectorData();
+			IECore::StringVectorDataPtr customAttributesData;
+			IECore::CompoundDataMap::const_iterator it = output->parameters().find( "custom_attributes" );
+			if( it == output->parameters().end() )
+			{
+				customAttributesData = new IECore::StringVectorData();
+			}
+			else
+			{
+				customAttributesData = IECore::runTimeCast<IECore::StringVectorData>( it->second );
```

Comment:
Ah, yes. I was actually thinking about that and then it just compiled and I thought it should be fine. Will update.